### PR TITLE
Add functions to support document deletion

### PR DESCRIPTION
**Pull request by amandaJayanetti**

### CODE REVIEW

File: R/RcppExports.R [hunk 1]
```diff
@@ -227,6 +227,38 @@ has_positions <- function(dbpath) {
...
+#' @examples
+#' \dontrun{
+#' dbpath<- c("path/to/database")
+#' docid <- 20
+#' deleteWrapper(dbpath,docid)
+#' }
+#' 
+#' @return none
+deleteWrapper <- function(dbpath, docid) {
+    invisible(.Call('RXapian_deleteWrapper', PACKAGE = 'RXapian', dbpath, docid))
```
Comment (jaylett) [inline R/RcppExports.R]: I've only just noticed, but you seem to have inconsistent indentation between different `.R` files: four spaces here, two in `R/DatabaseWrapper.R`. Worth tidying up.

File: src/DatabaseWrapper.cpp [hunk 2]
```diff
@@ -369,3 +369,43 @@ has_positions(SEXP dbpath)
...
+//' \dontrun{
+//' dbpath<- c("path/to/database")
+//' docid <- 20
+//' deleteWrapper(dbpath,docid)
+//' }
+//' 
+//' @return none
+// [[Rcpp::export]]
+void deleteWrapper(SEXP dbpath, SEXP docid){
+  std::string path=Rcpp::as<std::string>(dbpath);
```
Comment (jaylett) [inline src/DatabaseWrapper.cpp]: It'd be better to have indentation consistent across the entire file (previous function uses four spaces).

**Comment by jaylett** [comment]
Would it be possible to have one function, `xapian_delete`, that checked for either an integer or a string id? It's calling different overloaded functions in Xapian, but with the same name, so it might be easier for R users not to have to think about the different either?

**Comment by amandaJayanetti** [comment]
Since a class structure is not used, function overloading is not possible.
However, an extra boolean variable can be introduced to enable the users to
select which version of xapian_delete is required. Will that be a better
approach?

**Comment by jaylett** [comment]
No, I think that would be more fiddly to use.

**Pull request closed without merging**